source/develop/release-management/features/node/node-next-persistence.html.md

```diff
@@ -0,0 +1,90 @@
+---
+title: Next Gen Node RPM persistence
+category: feature
+authors: rbarry
+wiki_category: Feature|Next Gen Node RPM Persistence
+wiki_title: Feature/NextGenNodeRPMPersistence
+wiki_revision_count: 0
+wiki_last_updated: 2016-09-20
+feature_name: Reinstall/Persist RPMs after Node upgrades
+feature_modules: node persistence
+feature_status: WIP, 4.1 proposed feature
+---
+
+# Node RPM Persistence After Upgrades
+
+## Summary
+[oVirt Node NG](http://www.ovirt.org/develop/projects/node/4.0) is an advancement over vintage Node, especially that it allows users to install RPMs or customize the system through configuration management tools. Some of these changes are migrated over from /etc after upgrades, and some are kept persistenly in /var, but any packages which are installed on a running layer will be lost after upgrades and need re-installation. This presents some problems with vendor tooling, IPMI, utilities to collect statistics, and other add-ons.
+
+This feature presents a mechanism by which packages installed through yum/dnf can be saved and automatically re-applied when the OS is updated.
+
+### Owner
+* Name: Ryan Barry
+* Email: rbarry@redhat.com
+
+### Current Status
+* Status: Design
+* Last updated date: 20th Sep 2016
+
+### Benefit to oVirt
+
+This feature will greatly improve the user experience of Node by maintaining a consistent userland and system configuration after upgrades. This allows for Node customizations to "stick".
+
+### Dependencies
+None! We simply need to add an additional yum plugin to imgbased.
+
+## Detailed Description
+
+Node NG performs a number of actions when updates are applied.
+
+* A new layer is added
+* The new squashfs is extracted onto that layer
+* `/etc`/ and `/root` are rsynced to the new layer
+* UIDs and GIDs are synchronized across packages, in case a drift occurred inbetween images for UIDs/GIDs which are not fixed
+* A new bootloader entry is added and set as the default
+
+The existing osupdater logic can be extended (as the new root filesystem is already mounted) in order to install a package (or packages) into the new layer before it is booted for the first time.
```

Comment:
Yes, it would be a different plugin. I mentioned osupdater, since that has similar behavior.